### PR TITLE
Dscs 146

### PR DESCRIPTION
**Pull request by shuangzhou**
Hi Shaun,

This request includes fixes and unit tests for extracting correspondence index and container list from an EAD.
Could you please review these changes?

Thanks,
Shuang

### CODE REVIEW

File: src/amberdb/model/Node.java [hunk 1]
```diff
@@ -292,8 +292,8 @@ public Description getDescription(String fmt) {
             List<T> entries = new ArrayList<>();
             Iterator<Description> it = descriptions.iterator();
             while (it.hasNext()) {
-                Description next = it.next();
-                if (next.getType() != null && next.getType().equals(returnClass.getName())) {
+                Description next = it.next();  
+                if (next.getType() != null && (next.getType().equals(returnClass.getName()) || (next.getType().equals(returnClass.getSimpleName())))) {
```
Comment (scoen) [inline src/amberdb/model/Node.java]: I don't think the !=null is not protecting the 'or' (||) portion of this boolean statement.

I don't know if it needs to, but if getType() can return null there might be an issue

Comment (scoen) [inline src/amberdb/model/Node.java]: I don't think the !=null is protecting the 'or' (||) portion of this boolean statement. 

Is what i meant sorry

File: src/amberdb/model/builder/CollectionBuilder.java [hunk 2]
```diff
@@ -606,65 +607,74 @@ protected static void processCollection(Work collectionWork, InputStream in, Jso
     
     protected static void extractFeatures(EADWork collectionWork, JsonNode featuresCfg, XmlDocumentParser parser) {
         String basePath = featuresCfg.get(CFG_BASE).getTextValue();
-        String repeatablePath = featuresCfg.get(CFG_REPEATABLE_ELEMENTS).getTextValue();
-        Map<String, String> mapping = parser.getFieldsMap(parser.doc.getRootElement(), featuresCfg, basePath);
+        Nodes nodes = parser.getElementsByXPath(parser.getDocument(), basePath);
+        if (nodes.size() > 0) {
```
Comment (scoen) [inline src/amberdb/model/builder/CollectionBuilder.java]: could you indent the code in here ? Just formatting i know.

File: src/amberdb/model/Node.java (hunk 1 above)
Comment (scoen) [inline src/amberdb/model/Node.java]: ah i see what you mean now - carry on nothing to see here ...

File: src/amberdb/model/builder/CollectionBuilder.java (hunk 2 above)
Comment (shuangzhou) [inline src/amberdb/model/builder/CollectionBuilder.java]: it's been reformatted.

**Pull request merged**